src/test/java/frc/robot/util/FieldElementTest.java

```diff
@@ -0,0 +1,388 @@
+package frc.robot.util;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import edu.wpi.first.apriltag.AprilTag;
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.util.Units;
+import edu.wpi.first.wpilibj.DriverStation.Alliance;
+import frc.robot.constants.FieldConstants;
+import frc.robot.constants.VisionConstants;
+import frc.robot.util.Node.NodeType;
+
+/**
+ * Make sure that various field dimensions make sense.
+ * 
+ * <p>This test should run everytime someone builds the robot code.
+ * See https://docs.wpilib.org/en/stable/docs/software/wpilib-tools/robot-simulation/unit-testing.html
+ * 
+ * <p>To disable a test, annotate with Disabled
+ */
+
+public class FieldElementTest {
+    /** Field height is 26 feet 3.5 inches */
+    double heightField = Units.inchesToMeters(26 * 12 + 3.5);
+    /** Field width is 54 feet 3.25 inches */
+    double widthField = Units.inchesToMeters(54 * 12 + 3.25);
+
+    // all three grid units are 18 feet 0.5 inches (Game Manual)
+    double wholeGrid = 18.0 * 12 + 0.5;
+
+    // The outer Grid is 6 feet 3 inches (Game Manual)
+    double outerGrid = 6.0 * 12 + 3;
+    // The inner Grid is 5 feet 6 inches (Game Manual)
+    double innerGrid = 5.0 * 12 + 6;
+
+    // Outermost hybrid nodes are 2 feet 1.75 inches wide (Game Manual)
+    double outerHybrid = 2.0 * 12 + 1.75;
+    // other hybrid nodes are 1 feet 6.5 inches wide (Game Manual)
+    double innerHybrid = 1.0 * 12 + 6.5;
+    // width of the bar at the foot of grid (consistent value chosen from Game Manual dimensions)
+    // drawings call them Dividers
+    double widthBar = 3.5;
+
+    @BeforeEach
+    public void prepare() {
+    }
+  
+    @AfterEach
+    public void cleanup() {}
+
+    /**
+     * Check field dimensions
+     */
+    @Test
+    public void testFieldDimensions() {
+        // TODO: terminology: kFieldWidth is confusing because field is drawn landscape, so width != length
+        assertEquals(heightField, FieldConstants.kFieldWidth, 0.001);
```

Comment:
Here and throughout this PR: I do not understand why do we need tests that assert that constants have certain values. Is this to make harder to change constants? This seems counter-productive.